### ISSUE TITLE
`object-literal-sort-keys` false positive

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#3541]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const balls = {
    messages: [],
    messageValues: {}
};
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"object-literal-sort-keys": true
	}
}

```

#### Actual behavior

`ERROR: ... The key 'messageValues' is not sorted alphabetically`

#### Expected behavior

No error. I would expect this to behave consistently with `"member-ordering": [true, {"alphabetize": true}]`, which complains if `messageValues` doesn't come after `messages`.

**Comment by ajafff** [comment]
The comparison is case sensitive. Uppercase letters precede lowercase letters.

```js
"messages" > "messageValues" === true
"messages" < "messageValues" === false
```

**Comment by buu700** [comment]
Yep, I figured that much based on the result. It just seems like a bug that the alphabetization method isn't consistent across different rules.

**Comment by JoshuaKGoldberg** [comment]
> alphabetization method isn't consistent across different rules

Yes, this is odd. Feels like we should unify this?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
